### ISSUE TITLE
Unexpected error when importing `export =` with function declaration

### ISSUE THREAD

**Issue by OliverJAsh** [microsoft/TypeScript#36026]
**TypeScript Version:**  3.7.3

**Search Terms:** import export equals = declaration types typings module function declaration expression

**Code**

```ts
// This module can only be referenced with ECMAScript imports/exports by turning on the 'esModuleInterop' flag and referencing its default export.
import * as lib from "./lib";
```

```ts
// lib.d.ts
declare function md5(value: string, key?: string, raw?: boolean): string;

export = md5;
```

(Note: I have intentionally disabled `allowSyntheticDefaultImports` + `esModuleInterop`.)

#### Potential/known workarounds

If we use a function expression instead, the problem goes away:

```ts
// lib.d.ts
declare const md5: (value: string, key?: string, raw?: boolean) => string;

export = md5;
```

The problem also goes away if we add a `namespace`:

```ts
// lib.d.ts
declare function md5(value: string, key?: string, raw?: boolean): string;
declare namespace md5 {}

export = md5;
```

**Comment by RyanCavanaugh** [comment]
This is the intended behavior because `import * as x` can never produce a callable `x` per the ES spec. The other forms are allowed workarounds if you really intend to write some broken code 😉

**Comment by OliverJAsh** [comment]
Ah, thanks for clarifying that.

Given this is a valid CommonJS module definition, I'm wondering how we can import this into our TS project.

- We can't use `import md5 = require('blueimp-md5')` because we're targeting ES Modules
- We can't use `import * as md5 from 'blueimp-md5'` because ES Modules do not allow namespaces to be callable
- We can't use `import md5 from 'blueimp-md5'` because we haven't enabled `allowSyntheticDefaultImports`, nor do we want to (because [it can result in runtime exceptions](https://github.com/microsoft/TypeScript/issues/33954) and we want to adhere to the ES Modules spec)

Is this case, would you recommend using one of the workarounds I mentioned above, or is there something else we should consider?

**Comment by ishowta** [comment]
Why doesn't the error occur when namespace is declared? Even in that case, I think the same runtime error occured when the function is called.

**Comment by RyanCavanaugh** [comment]
Some people used downlevelers that made this incidentally work and took a hard dependency on that fact, so the namespace declaration is allowed as a workaround
